### ISSUE TITLE
Should Menu be refactored to use a "menu" role?

### ISSUE THREAD

**Issue by LMS007** [hypothesis/client#1735]
Currently the way our menu works is you can open it by pressing Enter, then tab around to change focus on the different items. The menu will close when you tab all the way past the end or back before the start of the items.

![Screen Shot 2020-01-29 at 3 30 02 PM](https://user-images.githubusercontent.com/3939074/73406825-3dfcb080-42ac-11ea-8164-1e4181d3f96c.png)
![Screen Shot 2020-01-29 at 3 30 27 PM](https://user-images.githubusercontent.com/3939074/73406845-4bb23600-42ac-11ea-8521-20c24556738a.png)

I'm wondering if this is sufficient to meet a11y requirements going forward and in looking over other menu examples I noticed they don't rely on focus state for the navigation.

https://www.w3.org/TR/wai-aria-practices/examples/

All of them seems to support basic keyboard navigation with the arrow keys, making a selection with Enter or space and then closing the menu with Esc. The core difference between our menu and these examples is we use a positive value for the `tabindex` which allows navigation to be done via changing focus state. 

1. Is our current menu sufficient as is for our a11y standards?

2. If we do want to refactor the component,  then we should pick an example that closely matches our UI. Examples that may fit:

https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-actions-active-descendant.html
https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-actions.html

Related to https://github.com/hypothesis/client/issues/1710

**Comment by LMS007** [comment]
A few more questions came up when looking at more lint issues.

Should the menu set focus on the first item when its opened?
Should it return focus to the button when its closed?
Should it perform this behavior for keyboard navigation only, or also mouse?

**Comment by robertknight** [comment]
Summarizing a discussion from the dev sync yesterday, since we use the `menu` ARIA role, our menus should be made to conform to the standard keyboard interaction and focus patterns described in the [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/#menu) guide.

**Comment by LMS007** [comment]
Related to
https://gist.github.com/robertknight/a2303f74f38a50736c20fafb7804e49e

**Comment by lyzadanger** [comment]
The `div` that contains a `Menu` component's menu items does now use a `menu` role, so I am going to close this.
